include/aikido/robot/ConcreteRobot.hpp

```diff
@@ -0,0 +1,262 @@
+#ifndef AIKIDO_ROBOT_CONCRETEROBOT_HPP_
+#define AIKIDO_ROBOT_CONCRETEROBOT_HPP_
+
+#include <chrono>
+#include <string>
+#include <unordered_map>
+#include <Eigen/Core>
+#include <dart/dart.hpp>
+#include <dart/dynamics/dynamics.hpp>
+#include "aikido/common/ExecutorThread.hpp"
+#include "aikido/common/RNG.hpp"
+#include "aikido/constraint/CollisionFree.hpp"
+#include "aikido/constraint/TSR.hpp"
+#include "aikido/control/TrajectoryExecutor.hpp"
+#include "aikido/planner/parabolic/ParabolicSmoother.hpp"
+#include "aikido/planner/parabolic/ParabolicTimer.hpp"
+#include "aikido/robot/Robot.hpp"
+#include "aikido/robot/util.hpp"
+#include "aikido/statespace/dart/MetaSkeletonStateSpace.hpp"
+#include "aikido/trajectory/Trajectory.hpp"
+
+namespace aikido {
+namespace robot {
+
+AIKIDO_DECLARE_POINTERS(ConcreteRobot)
+
+class ConcreteRobot : public Robot
+{
+public:
+  ConcreteRobot(
+      const std::string& name,
+      dart::dynamics::MetaSkeletonPtr robot,
+      statespace::dart::MetaSkeletonStateSpacePtr statespace,
+      bool simulation,
+      std::unique_ptr<aikido::common::RNG> rng,
```

Comment:
Should these be `aikido::common::UniqueRNGPtr` and `aikido::control::TrajectoryExecutorPtr`?